packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -97,6 +182,37 @@ internal struct MediaHandler {
     }
   }
 
+  @available(iOS 14, *)
+  private func handleVideo(itemProvider: NSItemProvider, completion: @escaping (Result<VideoInfo, Exception>) -> Void) {
+    itemProvider.loadFileRepresentation(forTypeIdentifier: UTType.movie.identifier) { [self] url, error in
+      do {
+        guard error == nil,
+              let videoUrl = url as? URL else {
+          return completion(.failure(FailedToReadVideoException().maybeCausedBy(error)))
+        }
+
+        let targetUrl = try generateUrl(withFileExtension: ".mov")
+        try VideoUtils.tryCopyingVideo(at: videoUrl, to: targetUrl)
+
+        guard let size = VideoUtils.readSizeFrom(url: targetUrl) else {
+          return completion(.failure(FailedToReadVideoSizeException()))
+        }
+
+        let duration = VideoUtils.readDurationFrom(url: targetUrl)
+
+        let result = VideoInfo(uri: targetUrl.absoluteString,
+                               width: size.width,
+                               height: size.height,
+                               duration: duration)
+        completion(.success(result))
+      } catch let exception as Exception {
+        return completion(.failure(exception))
+      } catch {
+        return completion(.failure(UnexpectedException(error)))
+      }
```

Comment:
This looks similar to the other `handleVideo` function body. Unify it.

Reply:
There are subtle differences in instructions inside `handleVideo` and `handleImage` - trials of unifying this introduced even more mess in helper functions so imo it's not worth it for now.